lib/acme/client/resources/authorization.rb

```diff
@@ -1,44 +1,86 @@
+# frozen_string_literal: true
+
 class Acme::Client::Resources::Authorization
-  HTTP01 = Acme::Client::Resources::Challenges::HTTP01
-  DNS01 = Acme::Client::Resources::Challenges::DNS01
-  TLSSNI01 = Acme::Client::Resources::Challenges::TLSSNI01
+  attr_reader :url, :expires, :status, :wildcard, :attributes
+
+  def self.arguments_from_response(response)
+    attributes = response.body
+    status = attributes.fetch('status')
+    expires =  Time.parse(attributes.fetch('expires'))
+    challenges = attributes.fetch('challenges')
+    wildcard = attributes.fetch('wildcard', false)
 
-  attr_reader :client, :uri, :domain, :status, :expires, :http01, :dns01, :tls_sni01
+    {
+      status: status,
+      expires: expires,
+      challenges: challenges,
+      wildcard: wildcard
+    }
+  end
 
-  def initialize(client, uri, response)
+  def initialize(client, **arguments)
     @client = client
-    @uri = uri
-    assign_attributes(response.body)
+    assign_attributes(arguments)
+  end
+
+  def deactivate
+    assign_attributes **@client.deactivate_authorization(url: url).to_h
+    true
+  end
+
+  def reload
+    assign_attributes **@client.authorization(url: url).to_h
+    true
+  end
+
+  def challenges
```

Comment:
Given the logic in `http01` and `dns01` I'd recommend introducing a `Challenges` object so that we can do

```ruby
def assign_attributes(...)
  @challenges = Challenges.new(challenges)
end
```

which would let you do

```ruby
authorization.challenges.http01
```

Reply:
Its a nice looking API but I'm not sure if it would bring value. In most scenario you will want to access a specific challenge. i.e.; `authorization.http01`